chain/bitcoind_conn.go

```diff
@@ -370,6 +430,135 @@ func (c *BitcoindConn) blockEventHandler() {
 	}
 }
 
+// blockEventHandlerRPC is a goroutine that uses the rpc client to check if we
+// have a new block every so often.
+func (c *BitcoindConn) blockEventHandlerRPC() {
+	defer c.wg.Done()
+
+	log.Info("Started polling for new bitcoind blocks via RPC.")
+	ticker := time.NewTicker(c.pollBlockTimer)
+
+	for {
+		select {
+		case <-ticker.C:
+			var (
+				blockHash *chainhash.Hash
+				err       error
+			)
+			// Every so many seconds, we poll to see if there's a new
+			// block.
+			blockHash, err = c.client.GetBestBlockHash()
+			if err != nil {
+				// Retry request once if we get an error.
+				time.Sleep(time.Second)
+				blockHash, err = c.client.GetBestBlockHash()
+				if err != nil {
+					log.Errorf("Unable to retrieve best block "+
+						"hash: %v", err)
+					continue
+				}
+			}
+
+			if blockHash == nil {
+				log.Errorf("Unable to retrieve best block "+
+					"hash: %v", err)
+				continue
+			}
+
+			c.rescanClientsMtx.Lock()
+			if c.rescanClients == nil || len(c.rescanClients) < 1 {
+				log.Errorf("No client added yet")
+				continue
+			}
+
+			// Grab one of the clients from the rescanClients map
+			// so we can look at its latest block height.
+			var client *BitcoindClient
+			idx := 0
+			for _, currClient := range c.rescanClients {
+				if idx > 0 {
+					break
+				}
+
+				client = currClient
+
+				idx += 1
+			}
```

Comment:
Yeah so the weird thing here is I encountered a case where... since we're registering a few different clients, the keys in the map are somewhat unpredictable. Like if there were 3 clients, I was running into a case where it only had a key of "3", and the other clients 1 and 2 (set by the counter above) hadn't been fully added to the map yet. So to make sure we grab a key that actually exists I added the loop since we're not sure what map keys are in the map atm. I'll add a comment here to explain this reasoning/choice!